src/example-types/facet.js

```diff
@@ -134,5 +135,77 @@ module.exports = {
           }),
         options
       ),
-    })),
+    }))
+    // results.options.name is ObjectId which need to stringify to get the correct missedValues
+    // stringify values to avoid the bug when values are numeric values
+    let missedValues = _.difference(
+      _.map(_.toString, values),
+      _.map(({ name }) => _.toString(name), results.options)
+    )
+
+    let getSelectValues = node =>
+      node.isMongoId ? _.map(ObjectID, values) : values
+
+    let matchSelectedValues = node => ({
+      $match: { _id: { $in: getSelectValues(node) } },
+    })
+
+    if (!_.isEmpty(missedValues)) {
+      //create a values hash map for filter noValuesOptions
+      let getValuesIndex = _.reduce(
+        (acc, value) => {
+          acc[value] = true
+          return acc
+        },
+        {},
+        values
+      )
+
+      let noValuesOptions = _.reject(
+        record => getValuesIndex[record.name],
+        results.options
+      )
+      let dropSize = noValuesOptions.length + values.length - node.size
+      if (dropSize > 0) noValuesOptions = _.drop(dropSize, noValuesOptions)
+
+      let searchValues = await search(
+        _.compact([
+          matchSelectedValues(node),
+          { $group: { _id: `$${node.field}`, count: { $sum: 1 } } },
+          ...(_.get('label', node)
+            ? [
+                {
+                  $lookup: {
+                    from: _.get('label.collection', node),
+                    as: 'label',
+                    localField: '_id',
+                    foreignField: _.get('label.foreignField', node),
+                  },
+                },
+                {
+                  $unwind: {
+                    path: '$label',
+                    preserveNullAndEmptyArrays: true,
+                  },
+                },
+              ]
+            : []),
```

Comment:
pretty sure this is the same as the first query - let's factor it out to reduce duplication here